pyliveradar.py

```
@@ -1,25 +1,107 @@
+            latest_file = links[-1]['href']
+            file_url = f"{url}{latest_file}"
+            logging.debug(f"Latest file URL: {file_url}")
+
+            # Download the radar data file
+            radar_response = requests.get(file_url, stream=True)
+            logging.debug(f"HTTP GET Response Status Code for file: {radar_response.status_code}")
+            radar_response.raise_for_status()
+            logging.debug("Downloaded radar data file successfully.")
+
+            # Save the file locally
+            output_path = os.path.join(output_dir, latest_file)
```

Ruff (0.11.9)

Undefined name `os`

(F821)